README.md

```diff
@@ -1,24 +1,314 @@
+internal (or embedded) and external.
+
+Inline includes the styles directly into the HTML element with the `style`
+attribute.
+
+```
+<p style="color: blue;"></p>
+```
+
+While you might see this sort of styling in something like the code for an HTML
+email, this is generally not best practice for two reasons. The primary reason
+is because it only affects that single element. If we want all paragraph
+elements on our page to appear blue, we would have to add that attribute to
+every element individually, which is inefficient and difficult to maintain into
+the future. That brings us to the second reason to avoid inline CSS: it breaks
+our principle of separation of content and presentation.
+
+Internal CSS is inside of `style` tags in the HTML document's `head` section.
+
+```
+<!DOCTYPE html>
+<html>
+  <head>
+    <style>
+      p { color: blue; }
+    </style>
+  </head>
+  <body>
+
+  <p>This is a paragraph.</p>
+
+  </body>
+</html>
+```
+
+This rule will display all paragraphs in this document as blue, which is a step
+up in scope from the inline styles that only apply to single elements. But this
+CSS will only apply to the single document. Other paragraph elements on other
+pages in the same website will be unaffected.
+
+If we want our CSS to carry across various pages, we can use an external
+stylesheet. This is a separate CSS file that we link in the `head` of HTML
+documents.
+
+```
+<!DOCTYPE html>
+<html>
+  <head>
+    <link rel="stylesheet" href="styles.css">
+  </head>
+  <body>
+
+  <p>This is a paragraph.</p>
+
+  </body>
+</html>
+```
+
+With the `link` tag, we can use the relation attribute to define the target as a
+stylesheet and the link source our CSS file that contains all the CSS we want to
+use on our site. This is by far the easiest way to link CSS to HTML and apply
+styles across all of our pages.
+
+## Utilize Various Types of CSS Selectors
+
+CSS gives us a wide range of ways that we can select elements on the page. Some
+of these you will use more than others, but it's a good idea to know them all so
+you can choose the right one when you need it.
+
+#### ID and Class Selectors
+
+ID selectors target all elements with a specific ID attribute value. The way we
+specify an ID selector in a CSS rule is is to follow the element name with a
+hash tag and then the ID attribute value we want to match.
+
+```
+p#introduction {
+  color: blue;
+}
+```
+
+In this case, the browser will look for a `p` element with the ID attribute
+"introduction" and apply the CSS to that element. ID selectors are useful when
+you want to give a single element on the page an unique identity and set it
+apart from everything else.
+
+Class selectors target all elements with a class attribute value matching the
+selector name. We specify a class selector using the period symbol followed by
+the name value.
+
+```
+p.alert {
+  color: red;
+}
+```
+
+The difference between IDs and classes is that IDs are meant for one element on
```

Comment:
My comment called this idea out earlier. I think it belongs-ish where I called it out.